script.js

```diff
@@ -52,4 +52,147 @@ function valutaConverter(event) {
   }
 }
 
-// updateing existing currency conversion rates with new rates form----------------
+// updateing existing currency conversion rates with new rates form
+// **********************************************
+// Week2 / An array of the currency rate objects
+// **********************************************
+
+const brandObjectArray = [
+  {
+    timestamp: Date.now(),
+    base: "EUR",
+    date: new Date().toLocaleDateString(),
+    rates: {},
+  },
+  {
+    timestamp: Date.now(),
+    base: "USD",
+    date: new Date().toLocaleDateString(),
+    rates: {},
+  },
+  {
+    timestamp: Date.now(),
+    base: "DKK",
+    date: new Date().toLocaleDateString(),
+    rates: {},
+  },
+];
+
+document
+  .getElementById("brand-currency-form-beta")
+  .addEventListener("submit", brandCurrencyBeta);
+function brandCurrencyBeta(event) {
+  event.preventDefault();
+
+  //getting and filling objects of array------------
+  const baseCurrency = document.getElementById("base-currency-beta").value;
+  const quoteName = document.getElementById("quote-name-beta").value;
+  const quoteRate = document.getElementById("quote-rate-beta").value;
+
+  const selectedCurrency = brandObjectArray.find(
+    (currency) => currency.base === baseCurrency
+  );
+
+  selectedCurrency.rates[quoteName] = +quoteRate;
+  document.getElementById("quote-rate-beta").value = "";
+  console.log(brandObjectArray);
+
+  //Calling function: Display currencies in table
+  currencyDisplayTable();
+}
+
+function currencyDisplayTable() {
+  const gridContainer = document.getElementById(
+    "currency-rates-grid-container"
+  );
+
+  // Clear the previous content in the grid container
+  gridContainer.innerHTML = "";
+
+  // Create a table element
+  const table = document.createElement("table");
+
+  // Create header row
+  const headerRow = table.createTHead().insertRow(0);
+  headerRow.insertCell(0).textContent = "Base Currency";
+  headerRow.insertCell(1).textContent = "Quote Currency";
+  headerRow.insertCell(2).textContent = "Rate";
+
+  // Create rows for each currency object
+  brandObjectArray.forEach((currency) => {
+    for (const quoteCurrency in currency.rates) {
+      const row = table.insertRow();
+      //creating 3 cells as there are 3 currencies: EUR, USD, DKK
+      row.insertCell(0).textContent = currency.base;
+      row.insertCell(1).textContent = quoteCurrency;
+      row.insertCell(2).textContent = currency.rates[quoteCurrency];
+    }
+  });
+
+  // Append the table to the grid container
+  gridContainer.appendChild(table);
+}
+
+//--- Second part of task WEEK 2   -------------------------------------------------------------
+// Getting a range for rates, searching for rates in brandObjectArray and show filterede data.
+let fromRate;
+let toRate;
+
+document
+  .getElementById("check-if-rate-exist")
+  .addEventListener("click", checkForRateExistancy);
```

Comment:
Thank you. Could you please tell me although it was 'click', it works and shows the table?

Reply:
`otherwise it will run the function even the user just click by mouse on the input` it listened of click of the mouse not clicking on the button, therefore the function was invoked everytime you clicked on anything inside of the form